src/tools/gophish_complete.py

```diff
@@ -46,46 +47,66 @@
 requests.packages.urllib3.disable_warnings()
 
 
-def get_campaigns(api):
-    """Return a dictionary containing all campaigns."""
+def get_campaign_id(campaign_name, campaigns):
+    """Get campaign id from campaign name.
+
+    Args:
+        campaign_name (string): Full campaign name.
+        campaigns (dict): Campaign id as key, campaign name as value.
+
+    Raises:
+        LookupError: When the campaign name is not found, raise exception.
+
+    Return:
+        Return campign id that corresponds to campaign name provided.
+    """
+    for campaign_id, name_value in campaigns.items():
+        if name_value == campaign_name:
+            return campaign_id
+
+    raise LookupError(f'Campaign Name "{campaign_name}" not found.')
+
+
+def get_campaigns(api, assessment_id):
+    """Return a dictionary containing all campaigns.
+
+    Args:
+        api (GoPhish API): Connection to GoPhish server via the API
+        assessment_id (string): Assessment identifier to get campaigns from.
+
+    Raises:
+        LookupError: If no campaigns are found for the assessment identifier, raise exception.
+
+    Returns:
+        dict: Campaign id as key, campaign name as value.
+    """
     allCampaigns = api.campaigns.get()
 
-    while True:
-        # Bandit complains about the input() function, but it is safe to
-        # use in Python 3, which is required by this project.
-        assessment_id = input("Enter Assessment ID: ")  # nosec
-        assessmentCampaigns = dict()
+    assessmentCampaigns = dict()
 
-        for campaign in allCampaigns:
-            if campaign.name.startswith(assessment_id):
-                assessmentCampaigns[campaign.id] = campaign.name
+    for campaign in allCampaigns:
+        if campaign.name.startswith(assessment_id):
+            assessmentCampaigns[campaign.id] = campaign.name
 
-        if len(assessmentCampaigns):
-            break
-        else:
-            logging.warning(
-                "No Campaigns found for Assessment {}".format(assessment_id)
-            )
-            print("Please try again...")
+    if len(assessmentCampaigns) == 0:
+        raise LookupError(f"No Campaigns found for Assessment {assessment_id}")
 
     return assessmentCampaigns
 
 
-def select_campaigns(campaigns):
+def select_campaign(campaigns):
     """Return the ID of a selected campaign."""
     print("Please select a Campaign ID:")
     print("\tID: Name")
 
     for key, name in campaigns.items():
-        print("\t" + str(key) + ": " + name)
+        print(f"\t {str(key)}: {name}")
```

Comment:
I don't think you still need the `str()` here now that this is an f-string, but please confirm that.

Also, I'd prefer the item variables here to be `id, name` to match the language used elsewhere in this code.

Reply:
Fixed in commit 1ab0c81.